File: Spieler.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Quizly
{
    public class Spieler
    {
        public int UserId { get; set; }
        public string Nickname { get; set; }
        public bool EmailConfirmed { get; set; }
        public DateTime? LastLogin { get; set; }

        public int GesamtPunkte { get; set; }

        public void AddPunkte(int punkte)
        {
            GesamtPunkte += punkte;
        }
    }
}

File: Android/app/src/main/java/com/quizly/android/data/model/Spieler.kt
package com.quizly.android.data.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/** Ported from Spieler.cs / spieler.py. */
@Serializable
data class Spieler(
    @SerialName("user_id")
    val userId: Int,
    val nickname: String,
    val fullname: String? = null,
    val email: String? = null,
    @SerialName("email_confirmed")
    val emailConfirmed: Boolean = false,
    @SerialName("last_login")
    val lastLogin: String? = null,
    val avatar: String? = null,
    // No confirmed source on the live API / Python model - kept nullable, only the
    // WPF client's Spieler.GesamtPunkte references a running point total.
    @SerialName("gesamt_punkte")
    val gesamtPunkte: Int? = null
)
